spec/unit/rust-crypto/rust-crypto.spec.ts

```diff
@@ -1532,34 +1534,47 @@ describe("RustCrypto", () => {
 
     describe("pinCurrentIdentity", () => {
         let rustCrypto: RustCrypto;
-        let olmMachine: Mocked<RustSdkCryptoJs.OlmMachine>;
 
-        beforeEach(() => {
-            olmMachine = {
-                getIdentity: jest.fn(),
-            } as unknown as Mocked<RustSdkCryptoJs.OlmMachine>;
-            rustCrypto = new RustCrypto(
-                logger,
-                olmMachine,
-                {} as MatrixClient["http"],
+        beforeEach(async () => {
+            const secretStorageCallbacks = {
+                getSecretStorageKey: async (keys: any, name: string) => {
+                    return [[...Object.keys(keys.keys)][0], new Uint8Array(32)];
+                },
+            } as SecretStorageCallbacks;
+            const secretStorage = new ServerSideSecretStorageImpl(new DummyAccountDataClient(), secretStorageCallbacks);
+            rustCrypto = await makeTestRustCrypto(
+                new MatrixHttpApi(new TypedEventEmitter<HttpApiEvent, HttpApiEventHandlerMap>(), {
+                    baseUrl: "http://server/",
+                    prefix: "",
+                    onlyData: true,
+                }),
                 TEST_USER,
                 TEST_DEVICE_ID,
-                {} as ServerSideSecretStorage,
-                {} as CryptoCallbacks,
+                secretStorage,
             );
         });
 
         it("throws an error for an unknown user", async () => {
-            await expect(rustCrypto.pinCurrentUserIdentity("@alice:example.com")).rejects.toThrow(
+            await expect(rustCrypto.pinCurrentUserIdentity("@other_user:example.com")).rejects.toThrow(
                 "Cannot pin identity of unknown user",
             );
         });
 
         it("throws an error for our own user", async () => {
-            const ownIdentity = new RustSdkCryptoJs.OwnUserIdentity();
```

Comment:
Isn't it more simple to just do            
```
 const ownIdentity = { } as unknown as Mocked<RustSdkCryptoJs.OwnUserIdentity>;
```
Instead of bringing all the key receiver/responder framework?

I added similar tests in this PR https://github.com/matrix-org/matrix-js-sdk/pull/4646/files

Reply:
The problem is that `pinCurrentUserIdentity` uses [`instanceof` to check whether it's an `OwnUserIdentity`](https://github.com/matrix-org/matrix-js-sdk/blob/develop/src/rust-crypto/rust-crypto.ts#L684), so it needs a real `OwnUserIdentity` object.